event_response_tool/event_response.py

```diff
@@ -116,34 +144,84 @@ def validate_inputs(event_response_inputs):
     ):
         alert.error("Please input AnalyzeRe password")
 
-    if (
-        event_response_inputs.event_weights_csv is None
-        or len(event_response_inputs.event_weights_csv) == 0
-    ):
-        alert.error(
-            "Please input the path of the CSV containing event weights"
-        )
-
-    # If layer_view CSV or portfolio_view UUID is not provided,
-    # check if analysis profile inputs are available
-    if (
-        event_response_inputs.layer_views_csv is None
-        or len(event_response_inputs.layer_views_csv) == 0
-    ) and (
-        event_response_inputs.portfolio_view_uuid is None
-        or len(event_response_inputs.portfolio_view_uuid) == 0
-    ):
-        if not (
-            event_response_inputs.old_analysis_profile_uuid
-            or (
-                event_response_inputs.total_number_of_events
-                and event_response_inputs.trial_count
+    # METHOD 1 - MIXTURE DISTRIBUTION METHOD
+    if event_response_inputs.mixture_distribution_method:
+        # Event weights CSV - required for creating weighted simgrid
+        if (
+            event_response_inputs.event_weights_csv is None
+            or len(event_response_inputs.event_weights_csv) == 0
+        ):
+            alert.error(
+                "Please input the path of the CSV containing event weights"
             )
+        if (
+            event_response_inputs.old_analysis_profile_uuid is None
+            or len(event_response_inputs.old_analysis_profile_uuid) == 0
         ):
             alert.error(
-                "Please provide inputs for either duplicating LayerViews or creating/updating Event Response Analysis Profile"
+                "Please provide the existing Analysis Profile UUID in 'old_analysis_profile_uuid' field for Mixture Distribution method"
+            )
+        try:
+            _ = int(event_response_inputs.max_trial_per_event)
+        except ValueError:
+            alert.error(
+                f"max_trial_per_event variable should be a valid integer: {event_response_inputs.max_trial_per_event}"
             )
 
+    # METHOD 2 - CONVOLUTION METHOD
+    if not event_response_inputs.mixture_distribution_method:
+        # If new Analysis Profile is to be created, check if trial count and
+        # and total number of events to be included in the catalog are provided
+        if event_response_inputs.create_analysis_profile:
+            if not (
+                event_response_inputs.total_number_of_events
+                and event_response_inputs.trial_count
+            ):
+                alert.error(
+                    "Please provide the total number of events and the trial count for creating a new Analysis Profile using the convolution method"
+                )
+
+        # If new Analysis Profile need not be created, check if
+        # an existing Analysis Profile UUID is provided
```

Comment:
looks like we're also checking that the event weights csv is provided

Reply:
Updated now